community/kernel/src/main/java/org/neo4j/kernel/impl/store/MetaDataStore.java

```diff
@@ -808,7 +808,7 @@ public boolean closedTransactionIdIsOnParWithOpenedTransactionId()
         return onPar;
     }
 
-    public void logRecords( final Logger msgLog )
+    public void logRecords( final Consumer<String> consumer )
```

Comment:
This is very generic and looks really strange.
Isn't printing all metadata records just the matter of giving it a `FormattedLogProvider.toOutputStream( System.out )` to `#logRecords()` ?